artemis/src/main/java/com/artemis/Manager.java

```diff
@@ -9,38 +11,41 @@
  * @author Arni Arent
  * @author Daan van Yperen
  */
-public abstract class Manager extends BaseSystem implements EntityObserver {
-
-	public void added(int entityId) {}
-	public void changed(int entityId) {}
-	public void deleted(int entityId) {}
-
-	@Override
-	public final void added(IntBag entities) {
-		int[] ids = entities.getData();
-		for (int i = 0, s = entities.size(); s > i; i++) {
-			added(ids[i]);
-		}
+public abstract class Manager extends BaseSystem {
+
+	public void added(Entity e) {}
+	public void deleted(Entity e) {}
+
+	protected void registerManager() {
+		world.getSystem(AspectSubscriptionManager.class)
+				.get(all())
+				.addSubscriptionListener(new EntitySubscription.SubscriptionListener() {
+					@Override
+					public void inserted(IntBag entities) {
+						added(entities);
+					}
+
+					@Override
+					public void removed(IntBag entities) {
+						deleted(entities);
+					}
+				});
 	}
 
-	@Override
-	public final void changed(IntBag entities) {
+	private void added(IntBag entities) {
 		int[] ids = entities.getData();
 		for (int i = 0, s = entities.size(); s > i; i++) {
-			changed(ids[i]);
+			added(world.getEntity(ids[i]));
```

Comment:
Perf impact isn't so bad, actually - it''s cheaper than processing systems; I left the optimization opportunity in there though.